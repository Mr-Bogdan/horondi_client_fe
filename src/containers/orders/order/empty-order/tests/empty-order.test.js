import React from 'react';
import * as reactRedux from 'react-redux';
import EmptyOrder from '../index';

let wrapper;

function spyOnSelector(lightMode) {
  jest.spyOn(reactRedux, 'useSelector').mockImplementation((cb) =>
    cb({
      Theme: { lightMode }
    })
  );
}

describe('Empty order component tests', () => {
  it('should match snapshot', () => {
    spyOnSelector(true);

    wrapper = shallow(<EmptyOrder title='' name='' buttonTitle='' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should cover other branches', () => {
    spyOnSelector(false);

    wrapper = shallow(<EmptyOrder title='' name='' buttonTitle='' />);
  });
});
