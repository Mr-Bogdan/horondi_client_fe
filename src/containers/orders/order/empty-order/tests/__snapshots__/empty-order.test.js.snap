// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty order component tests should match snapshot 1`] = `
<Fragment>
  <PathBack />
  <div
    className="makeStyles-root-4"
    data-cy=""
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      variant="h2"
    />
    <img
      alt=""
      className="makeStyles-image-3"
      src="./images/cart/empty-cart-light-theme-image.png"
    />
    <Link
      to="/catalog/:category"
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-1"
        variant="contained"
      />
    </Link>
  </div>
</Fragment>
`;
